src/runner/kafka.js

```diff
@@ -0,0 +1,58 @@
+// @flow
+import Kafka from 'no-kafka';
+import type { IRunner, Configuration, Logger, Consumer, IRegistry } from '../../types';
+
+class KafkaRunner implements IRunner {
+  config: Configuration;
+  logger: Logger;
+  registry: IRegistry;
+  consumer: Consumer;
+
+  constructor(config: Configuration, registry: IRegistry, logger: Logger) {
+    this.config = config;
+    this.registry = registry;
+    this.logger = logger;
+
+    this.consumer = new Kafka.GroupConsumer({
+      groupId: config.kafkaGroupId,
+      clientId: config.clientId,
+      connectionString: config.kafkaConnection,
+      codec: config.kafkaCodec,
+      logger: {
+        logLevel: config.logLevel,
+      },
+    });
+  }
+
+  receive = async (messages: Array<Object>, topic: string, partition: string) => {
+    for (const m of messages) { // eslint-disable-line no-restricted-syntax
+      let params: Object = {};
+      try {
+        // commit offset
+        params = JSON.parse(m.message.value.toString('utf8'));
+
+        this.registry.events.emit('runner_receive', topic, params);
+        await this.consumer.commitOffset({ topic, partition, offset: m.offset, metadata: 'optional' }); // eslint-disable-line
```

Comment:
do we need to implement an option of committing the offset after the task is finished?